src/main/java/com/gregtechceu/gtceu/api/cover/CoverBehavior.java

```
@@ -154,6 +154,17 @@ public boolean canConnectRedstone() {
         return false;
     }
 
+    @Override
+    public void copyConfig(CompoundTag nbt) {}
+
+    @Override
+    public Set<ItemStack> getItemsRequiredForPaste(CompoundTag nbt) {
```

Maybe throw an exception here to ensure it being overridden(because copying covers without needing items would be duping)

---

actually, all 3 of these methods should be abstract IMO

---

the cover item itself is required always, without the need to override this method
these are not abstract so that new covers don't have to implement them if they dont actually do any custom copying

---

discussed in discord, this list is *on top of* the cover item itself. Is resolved when a comment gets added clarifying that :)